pkg/services/ngalert/notifier/external_alertmanager.go

```diff
@@ -0,0 +1,159 @@
+package notifier
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/url"
+
+	httptransport "github.com/go-openapi/runtime/client"
+	"github.com/grafana/grafana/pkg/infra/log"
+	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	amclient "github.com/prometheus/alertmanager/api/v2/client"
+)
+
+type externalAlertmanager struct {
+	log           log.Logger
+	url           string
+	tenantID      string
+	orgID         int64
+	amClient      *amclient.AlertmanagerAPI
+	httpClient    *http.Client
+	defaultConfig string
+}
+
+type externalAlertmanagerConfig struct {
+	URL               string
+	TenantID          string
+	BasicAuthPassword string
+	DefaultConfig     string
+}
+
+func newExternalAlertmanager(cfg externalAlertmanagerConfig, orgID int64) (*externalAlertmanager, error) {
+	client := http.Client{
+		Transport: &roundTripper{
```

Comment:
We need this round-tripper to add basic auth and the X-Scope-OrgID header to requests made with the Alertmanager client.